### ISSUE TITLE
Feature Request: Google cloud authentication

### ISSUE THREAD

**Issue by NathanRVance** [gc3-uzh-ch/elasticluster#327]
When running Elasticluster from a google cloud instance, it's difficult without a GUI to accept Google's authentication prompt via a web browser. Cloud instances already have access to Google Cloud APIs, so would it be possible to configure Elasticluster to use that rather than a client id and secret?

**Comment by NathanRVance** [comment]
Just to be clear, when I provide the following entries in the configuration file:
> [cloud/google]
> provider=google
> gce_project_id=****REPLACE WITH YOUR PROJECT ID****
> gce_client_id=****REPLACE WITH YOUR CLIENT ID****
> gce_client_secret=****REPLACE WITH YOUR SECRET KEY****

and I start elasticluster on a google cloud computer (not my local computer) with:
> $ elasticluster start ipython -n mycluster

I get command line output stating that elasticluster is opening a particular URL in my webbrowser. Since my headnode is in a cloud computer, I don't have a webbrowser, authentication can't be completed, and the process hangs.

When setting up a cloud computer I've selected the option to give it unlimited access to GCE APIs. I can verify that I can spawn new instances from the command line using either google's REST API or the gcloud command line utility. Is it possible to bypass this web based authentication in elasticluster and use the authentication already in place?

**Comment by riccardomurri** [comment]
> When setting up a cloud computer I've selected the option to give it unlimited
> access to GCE APIs. I can verify that I can spawn new instances from the
> command line using either google's REST API or the gcloud command line
> utility. Is it possible to bypass this web based authentication in
> elasticluster and use the authentication already in place?

Sounds like a reasonable feature request, and something ElastiCluster
should indeed implement.

But I do not know Google's GCE API well enough to tell whether this is 
actually possible or not. Perhaps @mbookman can comment on the feasibility?

**Comment by mbookman** [comment]
[gcloud](https://cloud.google.com/sdk/gcloud/) now provides the ability for you to create credentials to be used by other applications.

The command to do this is:

    gcloud auth application-default login

The docs for the command are [here](https://cloud.google.com/sdk/gcloud/reference/auth/application-default/login).

This allows you to explicitly create [Application Default Credentials](https://developers.google.com/identity/protocols/application-default-credentials). That page has an [example of picking up ADC from Python](https://developers.google.com/identity/protocols/application-default-credentials#callingpython).

The gcloud command does take you through an authorization flow, but using ADC in Elasticluster would eliminate the need to create a Client ID and Client Secret and drop them into the configuration file.

**Comment by mbookman** [comment]
To the original problem, however set the `noauth_local_webserver` configuration value in the cloud provider section of `~/.elasticluster/config`:

```
# Create a cloud provider
[cloud/google-cloud]
provider=google
noauth_local_webserver=True
...
```

You'll still need to complete the authorization flow, but you get an URL to copy/paste into a browser, rather than Elasticluster trying to launch the browser directly.

**Comment by riccardomurri** [comment]
Fixed in PR #616 by @ndmckinley -- thanks a lot Nathan!
